<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8"/>
    <title>Test Report</title>
    <link href="assets/style.css" rel="stylesheet" type="text/css"/></head>
  <body onLoad="init()">
    <script>/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at http://mozilla.org/MPL/2.0/. */


function toArray(iter) {
    if (iter === null) {
        return null;
    }
    return Array.prototype.slice.call(iter);
}

function find(selector, elem) {
    if (!elem) {
        elem = document;
    }
    return elem.querySelector(selector);
}

function find_all(selector, elem) {
    if (!elem) {
        elem = document;
    }
    return toArray(elem.querySelectorAll(selector));
}

function sort_column(elem) {
    toggle_sort_states(elem);
    var colIndex = toArray(elem.parentNode.childNodes).indexOf(elem);
    var key;
    if (elem.classList.contains('numeric')) {
        key = key_num;
    } else if (elem.classList.contains('result')) {
        key = key_result;
    } else {
        key = key_alpha;
    }
    sort_table(elem, key(colIndex));
}

function show_all_extras() {
    find_all('.col-result').forEach(show_extras);
}

function hide_all_extras() {
    find_all('.col-result').forEach(hide_extras);
}

function show_extras(colresult_elem) {
    var extras = colresult_elem.parentNode.nextElementSibling;
    var expandcollapse = colresult_elem.firstElementChild;
    extras.classList.remove("collapsed");
    expandcollapse.classList.remove("expander");
    expandcollapse.classList.add("collapser");
}

function hide_extras(colresult_elem) {
    var extras = colresult_elem.parentNode.nextElementSibling;
    var expandcollapse = colresult_elem.firstElementChild;
    extras.classList.add("collapsed");
    expandcollapse.classList.remove("collapser");
    expandcollapse.classList.add("expander");
}

function show_filters() {
    var filter_items = document.getElementsByClassName('filter');
    for (var i = 0; i < filter_items.length; i++)
        filter_items[i].hidden = false;
}

function add_collapse() {
    // Add links for show/hide all
    var resulttable = find('table#results-table');
    var showhideall = document.createElement("p");
    showhideall.innerHTML = '<a href="javascript:show_all_extras()">Show all details</a> / ' +
                            '<a href="javascript:hide_all_extras()">Hide all details</a>';
    resulttable.parentElement.insertBefore(showhideall, resulttable);

    // Add show/hide link to each result
    find_all('.col-result').forEach(function(elem) {
        var extras = elem.parentNode.nextElementSibling;
        var expandcollapse = document.createElement("span");
        if (elem.innerHTML === 'Passed') {
            extras.classList.add("collapsed");
            expandcollapse.classList.add("expander");
        } else {
            expandcollapse.classList.add("collapser");
        }
        elem.appendChild(expandcollapse);

        elem.addEventListener("click", function(event) {
            if (event.currentTarget.parentNode.nextElementSibling.classList.contains("collapsed")) {
                show_extras(event.currentTarget);
            } else {
                hide_extras(event.currentTarget);
            }
        });
    })
}

function init () {
    reset_sort_headers();

    add_collapse();

    show_filters();

    toggle_sort_states(find('.initial-sort'));

    find_all('.sortable').forEach(function(elem) {
        elem.addEventListener("click",
                              function(event) {
                                  sort_column(elem);
                              }, false)
    });

};

function sort_table(clicked, key_func) {
    var rows = find_all('.results-table-row');
    var reversed = !clicked.classList.contains('asc');
    var sorted_rows = sort(rows, key_func, reversed);
    /* Whole table is removed here because browsers acts much slower
     * when appending existing elements.
     */
    var thead = document.getElementById("results-table-head");
    document.getElementById('results-table').remove();
    var parent = document.createElement("table");
    parent.id = "results-table";
    parent.appendChild(thead);
    sorted_rows.forEach(function(elem) {
        parent.appendChild(elem);
    });
    document.getElementsByTagName("BODY")[0].appendChild(parent);
}

function sort(items, key_func, reversed) {
    var sort_array = items.map(function(item, i) {
        return [key_func(item), i];
    });
    var multiplier = reversed ? -1 : 1;

    sort_array.sort(function(a, b) {
        var key_a = a[0];
        var key_b = b[0];
        return multiplier * (key_a >= key_b ? 1 : -1);
    });

    return sort_array.map(function(item) {
        var index = item[1];
        return items[index];
    });
}

function key_alpha(col_index) {
    return function(elem) {
        return elem.childNodes[1].childNodes[col_index].firstChild.data.toLowerCase();
    };
}

function key_num(col_index) {
    return function(elem) {
        return parseFloat(elem.childNodes[1].childNodes[col_index].firstChild.data);
    };
}

function key_result(col_index) {
    return function(elem) {
        var strings = ['Error', 'Failed', 'Rerun', 'XFailed', 'XPassed',
                       'Skipped', 'Passed'];
        return strings.indexOf(elem.childNodes[1].childNodes[col_index].firstChild.data);
    };
}

function reset_sort_headers() {
    find_all('.sort-icon').forEach(function(elem) {
        elem.parentNode.removeChild(elem);
    });
    find_all('.sortable').forEach(function(elem) {
        var icon = document.createElement("div");
        icon.className = "sort-icon";
        icon.textContent = "vvv";
        elem.insertBefore(icon, elem.firstChild);
        elem.classList.remove("desc", "active");
        elem.classList.add("asc", "inactive");
    });
}

function toggle_sort_states(elem) {
    //if active, toggle between asc and desc
    if (elem.classList.contains('active')) {
        elem.classList.toggle('asc');
        elem.classList.toggle('desc');
    }

    //if inactive, reset all other functions and add ascending active
    if (elem.classList.contains('inactive')) {
        reset_sort_headers();
        elem.classList.remove('inactive');
        elem.classList.add('active');
    }
}

function is_all_rows_hidden(value) {
  return value.hidden == false;
}

function filter_table(elem) {
    var outcome_att = "data-test-result";
    var outcome = elem.getAttribute(outcome_att);
    class_outcome = outcome + " results-table-row";
    var outcome_rows = document.getElementsByClassName(class_outcome);

    for(var i = 0; i < outcome_rows.length; i++){
        outcome_rows[i].hidden = !elem.checked;
    }

    var rows = find_all('.results-table-row').filter(is_all_rows_hidden);
    var all_rows_hidden = rows.length == 0 ? true : false;
    var not_found_message = document.getElementById("not-found-message");
    not_found_message.hidden = !all_rows_hidden;
}
</script>
    <p>Report generated on 07-Sep-2017 at 17:23:27 by<a href="https://pypi.python.org/pypi/pytest-html"> pytest-html</a> v1.15.2</p>
    <h2>Environment</h2>
    <table id="environment">
      <tr>
        <td>Packages</td>
        <td>{&apos;pytest&apos;: &apos;3.2.1&apos;, &apos;py&apos;: &apos;1.4.34&apos;, &apos;pluggy&apos;: &apos;0.4.0&apos;}</td></tr>
      <tr>
        <td>Platform</td>
        <td>Darwin-16.5.0-x86_64-i386-64bit</td></tr>
      <tr>
        <td>Plugins</td>
        <td>{&apos;metadata&apos;: &apos;1.5.0&apos;, &apos;json&apos;: &apos;0.4.0&apos;, &apos;html&apos;: &apos;1.15.2&apos;}</td></tr>
      <tr>
        <td>Python</td>
        <td>3.6.1</td></tr></table>
    <h2>Summary</h2>
    <p>31 tests ran in 7.84 seconds. </p>
    <p class="filter" hidden="true">(Un)check the boxes to filter the results.</p><input checked="true" class="filter" data-test-result="passed" hidden="true" name="filter_checkbox" onChange="filter_table(this)" type="checkbox"/><span class="passed">6 passed</span>, <input checked="true" class="filter" data-test-result="skipped" disabled="true" hidden="true" name="filter_checkbox" onChange="filter_table(this)" type="checkbox"/><span class="skipped">0 skipped</span>, <input checked="true" class="filter" data-test-result="failed" hidden="true" name="filter_checkbox" onChange="filter_table(this)" type="checkbox"/><span class="failed">25 failed</span>, <input checked="true" class="filter" data-test-result="error" disabled="true" hidden="true" name="filter_checkbox" onChange="filter_table(this)" type="checkbox"/><span class="error">0 errors</span>, <input checked="true" class="filter" data-test-result="xfailed" disabled="true" hidden="true" name="filter_checkbox" onChange="filter_table(this)" type="checkbox"/><span class="xfailed">0 expected failures</span>, <input checked="true" class="filter" data-test-result="xpassed" disabled="true" hidden="true" name="filter_checkbox" onChange="filter_table(this)" type="checkbox"/><span class="xpassed">0 unexpected passes</span>
    <h2>Results</h2>
    <table id="results-table">
      <thead id="results-table-head">
        <tr>
          <th class="sortable result initial-sort" col="result">Result</th>
          <th class="sortable" col="name">Test</th>
          <th class="sortable numeric" col="duration">Duration</th>
          <th>Links</th></tr>
        <tr hidden="true" id="not-found-message">
          <th colspan="4">No results found. Try to check the filters</th></tr></thead>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_simple2</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.test_connection.TestBasicConnection object at 0x104961dd8&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x10495a0b8&gt;<br/><br/>    def test_connect_simple2(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Sends a USER command followed by a NICK command, and expects<br/>            to receive, at least, a RPL_WELCOME reply.<br/>            &quot;&quot;&quot;<br/>    <br/>        client = irc_session.get_client(nodelay = True)<br/>    <br/>        client.send_raw([&quot;USER user1 * * :User One\r\n&quot;,<br/>                         &quot;NICK user1\r\n&quot;],<br/>                        wait=0.05)<br/>    <br/>        irc_session.get_reply(client, expect_code = replies.RPL_WELCOME, expect_nick=&quot;user1&quot;, expect_nparams = 1,<br/>&gt;                             long_param_re= &quot;Welcome to the Internet Relay Network user1!user1@.*&quot;)<br/><br/>tests/chirc/tests/test_connection.py:36: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/>tests/chirc/tests/common.py:356: in get_reply<br/>    self.verify_reply(msg, expect_code, expect_nick, expect_nparams, expect_short_params, long_param_re, long_param_values)<br/>tests/chirc/tests/common.py:431: in verify_reply<br/>    long_param_re = long_param_re, long_param_values = long_param_values)<br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x10495a0b8&gt;<br/>msg = &lt;chirc.types.IRCMessage object at 0x10499ff28&gt;, expect_prefix = True<br/>expect_cmd = &#x27;001&#x27;, expect_nparams = 2, expect_short_params = [&#x27;user1&#x27;]<br/>long_param_re = &#x27;Welcome to the Internet Relay Network user1!user1@.*&#x27;<br/>long_param_values = None<br/><br/>    def verify_message(self, msg, expect_prefix = None, expect_cmd = None,<br/>                      expect_nparams = None, expect_short_params = None,<br/>                      long_param_re = None, long_param_values = None):<br/>    <br/>        if expect_prefix != None and expect_prefix:<br/>&gt;           assert msg.prefix is not None, &quot;Expected a prefix, but got none.\nMessage: {}&quot;.format(msg.raw(bookends=True))<br/><span class="error">E           AssertionError: Expected a prefix, but got none.</span><br/><span class="error">E           Message: |||error, nickname already in use|||</span><br/><br/>tests/chirc/tests/common.py:380: AssertionError<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_both_messages_at_once</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.common.IRCSession object at 0x104aad898&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104aadbe0&gt;, expect_code = &#x27;001&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>&gt;           msg = client.get_message()<br/><br/>tests/chirc/tests/common.py:340: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.client.ChircClient object at 0x104aadbe0&gt;<br/><br/>    def get_message(self):<br/>        msg = self.client.read_until(str.encode(&quot;\r\n&quot;), timeout=self.msg_timeout)<br/>        msg = msg.decode()<br/>        if msg[-2:] != &quot;\r\n&quot;:<br/>&gt;           raise ReplyTimeoutException(msg)<br/><span class="error">E           chirc.types.ReplyTimeoutException</span><br/><br/>tests/chirc/client.py:37: ReplyTimeoutException<br/><br/>During handling of the above exception, another exception occurred:<br/><br/>self = &lt;chirc.tests.test_connection.TestBasicConnection object at 0x1049bbf60&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104aad898&gt;<br/><br/>    def test_connect_both_messages_at_once(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Sends a NICK and USER command in the same TCP packet. This tests that<br/>            the server doesn&#x27;t (incorrectly) assume that calling recv() will always<br/>            return a single message.<br/>    <br/>            Note: TCP will sometimes do whatever TCP wants, which means this message<br/>            could get partitioned.<br/>            &quot;&quot;&quot;<br/>    <br/>        client = irc_session.get_client(nodelay = True)<br/>    <br/>        client.send_raw(&quot;NICK user1\r\nUSER user1 * * :User One\r\n&quot;)<br/>    <br/>&gt;       irc_session.get_reply(client, expect_code = replies.RPL_WELCOME, expect_nick=&quot;user1&quot;, expect_nparams = 1)<br/><br/>tests/chirc/tests/test_connection.py:53: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104aad898&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104aadbe0&gt;, expect_code = &#x27;001&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>            msg = client.get_message()<br/>    <br/>            if expect_timeout:<br/>                pytest.fail(&quot;Was not expecting a reply, but got one:\n&quot; + msg.raw(bookends=True))<br/>        except EOFError:<br/>            pytest.fail(&quot;Server closed connection unexpectedly. Possible segfault in server?&quot;)<br/>        except ReplyTimeoutException as rte:<br/>            if expect_timeout:<br/>                return None<br/>    <br/>            if len(rte.bytes_received) == 0:<br/>                failmsg = &quot;Expected a reply but got none (no bytes received)&quot;<br/>            else:<br/>                failmsg = &quot;Expected a reply but did not get valid reply terminated with \\r\\n. Bytes received:\n|||{}|||&quot;.format(rte.bytes_received)<br/>&gt;           pytest.fail(failmsg)<br/><span class="error">E           Failed: Expected a reply but got none (no bytes received)</span><br/><br/>tests/chirc/tests/common.py:354: Failed<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_partitioned_messages1</td>
          <td class="col-duration">0.21</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.common.IRCSession object at 0x104b7a400&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104b7a978&gt;, expect_code = &#x27;001&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>&gt;           msg = client.get_message()<br/><br/>tests/chirc/tests/common.py:340: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.client.ChircClient object at 0x104b7a978&gt;<br/><br/>    def get_message(self):<br/>        msg = self.client.read_until(str.encode(&quot;\r\n&quot;), timeout=self.msg_timeout)<br/>        msg = msg.decode()<br/>        if msg[-2:] != &quot;\r\n&quot;:<br/>&gt;           raise ReplyTimeoutException(msg)<br/><span class="error">E           chirc.types.ReplyTimeoutException</span><br/><br/>tests/chirc/client.py:37: ReplyTimeoutException<br/><br/>During handling of the above exception, another exception occurred:<br/><br/>self = &lt;chirc.tests.test_connection.TestBasicConnection object at 0x104b7a160&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104b7a400&gt;<br/><br/>    def test_connect_partitioned_messages1(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Sends a NICK and USER command, but partitioned at a point other than<br/>            at the \r\n terminator.<br/>    <br/>            Note: TCP will sometimes do whatever TCP wants, which means this message<br/>            could get partitioned in other ways.<br/>            &quot;&quot;&quot;<br/>    <br/>        client = irc_session.get_client(nodelay = True)<br/>    <br/>        client.send_raw([&quot;NICK &quot;,<br/>                         &quot;user1\r\nUSER user1 * * :User One\r\n&quot;],<br/>                        wait=0.05)<br/>    <br/>&gt;       irc_session.get_reply(client, expect_code = replies.RPL_WELCOME, expect_nick=&quot;user1&quot;, expect_nparams = 1)<br/><br/>tests/chirc/tests/test_connection.py:70: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104b7a400&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104b7a978&gt;, expect_code = &#x27;001&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>            msg = client.get_message()<br/>    <br/>            if expect_timeout:<br/>                pytest.fail(&quot;Was not expecting a reply, but got one:\n&quot; + msg.raw(bookends=True))<br/>        except EOFError:<br/>            pytest.fail(&quot;Server closed connection unexpectedly. Possible segfault in server?&quot;)<br/>        except ReplyTimeoutException as rte:<br/>            if expect_timeout:<br/>                return None<br/>    <br/>            if len(rte.bytes_received) == 0:<br/>                failmsg = &quot;Expected a reply but got none (no bytes received)&quot;<br/>            else:<br/>                failmsg = &quot;Expected a reply but did not get valid reply terminated with \\r\\n. Bytes received:\n|||{}|||&quot;.format(rte.bytes_received)<br/>&gt;           pytest.fail(failmsg)<br/><span class="error">E           Failed: Expected a reply but got none (no bytes received)</span><br/><br/>tests/chirc/tests/common.py:354: Failed<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_partitioned_messages2</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.test_connection.TestBasicConnection object at 0x104aad160&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104aad5f8&gt;<br/><br/>    def test_connect_partitioned_messages2(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Sends a NICK and USER command, but partitioned at a point other than<br/>            at the \r\n terminator.<br/>    <br/>            Note: TCP will sometimes do whatever TCP wants, which means this message<br/>            could get partitioned in other ways.<br/>            &quot;&quot;&quot;<br/>    <br/>        client = irc_session.get_client(nodelay = True)<br/>    <br/>        client.send_raw([&quot;NICK user1\r\nUSER user1 &quot;,<br/>                         &quot;* * :User One\r\n&quot;],<br/>                        wait=0.05)<br/>    <br/>&gt;       irc_session.get_reply(client, expect_code = replies.RPL_WELCOME, expect_nick=&quot;user1&quot;, expect_nparams = 1)<br/><br/>tests/chirc/tests/test_connection.py:87: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/>tests/chirc/tests/common.py:356: in get_reply<br/>    self.verify_reply(msg, expect_code, expect_nick, expect_nparams, expect_short_params, long_param_re, long_param_values)<br/>tests/chirc/tests/common.py:431: in verify_reply<br/>    long_param_re = long_param_re, long_param_values = long_param_values)<br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104aad5f8&gt;<br/>msg = &lt;chirc.types.IRCMessage object at 0x104b2be10&gt;, expect_prefix = True<br/>expect_cmd = &#x27;001&#x27;, expect_nparams = 2, expect_short_params = [&#x27;user1&#x27;]<br/>long_param_re = None, long_param_values = None<br/><br/>    def verify_message(self, msg, expect_prefix = None, expect_cmd = None,<br/>                      expect_nparams = None, expect_short_params = None,<br/>                      long_param_re = None, long_param_values = None):<br/>    <br/>        if expect_prefix != None and expect_prefix:<br/>&gt;           assert msg.prefix is not None, &quot;Expected a prefix, but got none.\nMessage: {}&quot;.format(msg.raw(bookends=True))<br/><span class="error">E           AssertionError: Expected a prefix, but got none.</span><br/><span class="error">E           Message: |||error, nickname already in use|||</span><br/><br/>tests/chirc/tests/common.py:380: AssertionError<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_partitioned_messages3</td>
          <td class="col-duration">0.21</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.common.IRCSession object at 0x104d9deb8&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104e05278&gt;, expect_code = &#x27;001&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>&gt;           msg = client.get_message()<br/><br/>tests/chirc/tests/common.py:340: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.client.ChircClient object at 0x104e05278&gt;<br/><br/>    def get_message(self):<br/>        msg = self.client.read_until(str.encode(&quot;\r\n&quot;), timeout=self.msg_timeout)<br/>        msg = msg.decode()<br/>        if msg[-2:] != &quot;\r\n&quot;:<br/>&gt;           raise ReplyTimeoutException(msg)<br/><span class="error">E           chirc.types.ReplyTimeoutException</span><br/><br/>tests/chirc/client.py:37: ReplyTimeoutException<br/><br/>During handling of the above exception, another exception occurred:<br/><br/>self = &lt;chirc.tests.test_connection.TestBasicConnection object at 0x104d9d3c8&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104d9deb8&gt;<br/><br/>    def test_connect_partitioned_messages3(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Sends a NICK and USER command, but partitioned at a point other than<br/>            at the \r\n terminator.<br/>    <br/>            Note: TCP will sometimes do whatever TCP wants, which means this message<br/>            could get partitioned in other ways.<br/>            &quot;&quot;&quot;<br/>    <br/>        client = irc_session.get_client(nodelay = True)<br/>    <br/>        client.send_raw([&quot;NICK user1&quot;,<br/>                         &quot;\r\nUSER user1 * * :User One\r\n&quot;],<br/>                        wait=0.05)<br/>    <br/>&gt;       irc_session.get_reply(client, expect_code = replies.RPL_WELCOME, expect_nick=&quot;user1&quot;, expect_nparams = 1)<br/><br/>tests/chirc/tests/test_connection.py:104: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104d9deb8&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104e05278&gt;, expect_code = &#x27;001&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>            msg = client.get_message()<br/>    <br/>            if expect_timeout:<br/>                pytest.fail(&quot;Was not expecting a reply, but got one:\n&quot; + msg.raw(bookends=True))<br/>        except EOFError:<br/>            pytest.fail(&quot;Server closed connection unexpectedly. Possible segfault in server?&quot;)<br/>        except ReplyTimeoutException as rte:<br/>            if expect_timeout:<br/>                return None<br/>    <br/>            if len(rte.bytes_received) == 0:<br/>                failmsg = &quot;Expected a reply but got none (no bytes received)&quot;<br/>            else:<br/>                failmsg = &quot;Expected a reply but did not get valid reply terminated with \\r\\n. Bytes received:\n|||{}|||&quot;.format(rte.bytes_received)<br/>&gt;           pytest.fail(failmsg)<br/><span class="error">E           Failed: Expected a reply but got none (no bytes received)</span><br/><br/>tests/chirc/tests/common.py:354: Failed<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_partitioned_messages4</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.test_connection.TestBasicConnection object at 0x104e127b8&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104950c50&gt;<br/><br/>    def test_connect_partitioned_messages4(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Sends a NICK and USER command, but partitioned at a point other than<br/>            at the \r\n terminator.<br/>    <br/>            Note: TCP will sometimes do whatever TCP wants, which means this message<br/>            could get partitioned in other ways.<br/>            &quot;&quot;&quot;<br/>    <br/>        client = irc_session.get_client(nodelay = True)<br/>    <br/>        client.send_raw([&quot;NICK user1\r\nUSER user1 * * :User One&quot;,<br/>                         &quot;\r\n&quot;],<br/>                        wait=0.05)<br/>    <br/>&gt;       irc_session.get_reply(client, expect_code = replies.RPL_WELCOME, expect_nick=&quot;user1&quot;, expect_nparams = 1)<br/><br/>tests/chirc/tests/test_connection.py:121: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/>tests/chirc/tests/common.py:356: in get_reply<br/>    self.verify_reply(msg, expect_code, expect_nick, expect_nparams, expect_short_params, long_param_re, long_param_values)<br/>tests/chirc/tests/common.py:431: in verify_reply<br/>    long_param_re = long_param_re, long_param_values = long_param_values)<br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104950c50&gt;<br/>msg = &lt;chirc.types.IRCMessage object at 0x104e122b0&gt;, expect_prefix = True<br/>expect_cmd = &#x27;001&#x27;, expect_nparams = 2, expect_short_params = [&#x27;user1&#x27;]<br/>long_param_re = None, long_param_values = None<br/><br/>    def verify_message(self, msg, expect_prefix = None, expect_cmd = None,<br/>                      expect_nparams = None, expect_short_params = None,<br/>                      long_param_re = None, long_param_values = None):<br/>    <br/>        if expect_prefix != None and expect_prefix:<br/>&gt;           assert msg.prefix is not None, &quot;Expected a prefix, but got none.\nMessage: {}&quot;.format(msg.raw(bookends=True))<br/><span class="error">E           AssertionError: Expected a prefix, but got none.</span><br/><span class="error">E           Message: |||error, nickname already in use|||</span><br/><br/>tests/chirc/tests/common.py:380: AssertionError<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_partitioned_messages7</td>
          <td class="col-duration">0.70</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.common.IRCSession object at 0x104e05668&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104e055f8&gt;, expect_code = &#x27;001&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>&gt;           msg = client.get_message()<br/><br/>tests/chirc/tests/common.py:340: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.client.ChircClient object at 0x104e055f8&gt;<br/><br/>    def get_message(self):<br/>        msg = self.client.read_until(str.encode(&quot;\r\n&quot;), timeout=self.msg_timeout)<br/>        msg = msg.decode()<br/>        if msg[-2:] != &quot;\r\n&quot;:<br/>&gt;           raise ReplyTimeoutException(msg)<br/><span class="error">E           chirc.types.ReplyTimeoutException</span><br/><br/>tests/chirc/client.py:37: ReplyTimeoutException<br/><br/>During handling of the above exception, another exception occurred:<br/><br/>self = &lt;chirc.tests.test_connection.TestBasicConnection object at 0x104e19710&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104e05668&gt;<br/><br/>    def test_connect_partitioned_messages7(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Sends a NICK and USER command, but partitioned at a point other than<br/>            at the \r\n terminator.<br/>    <br/>            Note: TCP will sometimes do whatever TCP wants, which means this message<br/>            could get partitioned in other ways.<br/>            &quot;&quot;&quot;<br/>    <br/>        client = irc_session.get_client(nodelay = True)<br/>    <br/>        client.send_raw([&quot;NI&quot;,<br/>                         &quot;CK &quot;,<br/>                         &quot;user1\r&quot;,<br/>                         &quot;\n&quot;,<br/>                         &quot;USER user&quot;,<br/>                         &quot;1 * * &quot;,<br/>                         &quot;:Us&quot;,<br/>                         &quot;er &quot;,<br/>                         &quot;One&quot;,<br/>                         &quot;\r&quot;,<br/>                         &quot;\n&quot;],<br/>                        wait=0.05)<br/>    <br/>&gt;       irc_session.get_reply(client, expect_code = replies.RPL_WELCOME, expect_nick=&quot;user1&quot;, expect_nparams = 1)<br/><br/>tests/chirc/tests/test_connection.py:184: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104e05668&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104e055f8&gt;, expect_code = &#x27;001&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>            msg = client.get_message()<br/>    <br/>            if expect_timeout:<br/>                pytest.fail(&quot;Was not expecting a reply, but got one:\n&quot; + msg.raw(bookends=True))<br/>        except EOFError:<br/>            pytest.fail(&quot;Server closed connection unexpectedly. Possible segfault in server?&quot;)<br/>        except ReplyTimeoutException as rte:<br/>            if expect_timeout:<br/>                return None<br/>    <br/>            if len(rte.bytes_received) == 0:<br/>                failmsg = &quot;Expected a reply but got none (no bytes received)&quot;<br/>            else:<br/>                failmsg = &quot;Expected a reply but did not get valid reply terminated with \\r\\n. Bytes received:\n|||{}|||&quot;.format(rte.bytes_received)<br/>&gt;           pytest.fail(failmsg)<br/><span class="error">E           Failed: Expected a reply but got none (no bytes received)</span><br/><br/>tests/chirc/tests/common.py:354: Failed<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_nick_user_parsing1</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.test_connection.TestBasicConnection object at 0x104e192e8&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104e19630&gt;<br/><br/>    def test_connect_nick_user_parsing1(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Tests that the server is actually parsing the NICK and USER parameters<br/>            (as opposed to just hardcoding &quot;nick1&quot; and &quot;user1&quot; which is used in<br/>            other tests.<br/>            &quot;&quot;&quot;<br/>    <br/>        client = irc_session.get_client(nodelay = True)<br/>    <br/>        client.send_raw([&quot;NICK nick42\r\n&quot;,<br/>                         &quot;USER user42 * * :User Forty Two\r\n&quot;],<br/>                        wait=0.05)<br/>    <br/>        irc_session.get_reply(client, expect_code = replies.RPL_WELCOME, expect_nick=&quot;nick42&quot;, expect_nparams = 1,<br/>&gt;                             long_param_re= &quot;Welcome to the Internet Relay Network nick42!user42@.*&quot;)<br/><br/>tests/chirc/tests/test_connection.py:200: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/>tests/chirc/tests/common.py:356: in get_reply<br/>    self.verify_reply(msg, expect_code, expect_nick, expect_nparams, expect_short_params, long_param_re, long_param_values)<br/>tests/chirc/tests/common.py:431: in verify_reply<br/>    long_param_re = long_param_re, long_param_values = long_param_values)<br/>tests/chirc/tests/common.py:404: in verify_message<br/>    self._assert_is_not_none(match, &quot;|||%s||| &lt;-- Long parameter does not match regular expression: %s&quot; % (lp, lpre), irc_msg = msg)<br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104e19630&gt;, a = None<br/>explanation = &#x27;|||:Welcome to the Internet Relay Network nick42!nick42@localhost||| &lt;-- Long parameter does not match regular expres...2@.*$\n\nMESSAGE: |||:MacBook-Pro-5.local 001 nick42 :Welcome to the Internet Relay Network nick42!nick42@localhost|||&#x27;<br/>irc_msg = &lt;chirc.types.IRCMessage object at 0x104e05f98&gt;<br/><br/>    def _assert_is_not_none(self, a, explanation, irc_msg = None):<br/>        if irc_msg is not None:<br/>            explanation = explanation + &quot;\n\nMESSAGE: {}&quot;.format(irc_msg.raw(bookends=True))<br/>    <br/>&gt;       assert a is not None, explanation<br/><span class="error">E       AssertionError: |||:Welcome to the Internet Relay Network nick42!nick42@localhost||| &lt;-- Long parameter does not match regular expression: ^:Welcome to the Internet Relay Network nick42!user42@.*$</span><br/><span class="error">E       </span><br/><span class="error">E       MESSAGE: |||:MacBook-Pro-5.local 001 nick42 :Welcome to the Internet Relay Network nick42!nick42@localhost|||</span><br/><br/>tests/chirc/tests/common.py:56: AssertionError<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_nick_user_parsing2</td>
          <td class="col-duration">0.22</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.test_connection.TestBasicConnection object at 0x104e17208&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104e17a20&gt;<br/><br/>    def test_connect_nick_user_parsing2(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Tests that the server is actually parsing the NICK and USER parameters<br/>            (as opposed to just hardcoding &quot;nick1&quot; and &quot;user1&quot; which is used in<br/>            other tests. Additionally, partitions the messages.<br/>            &quot;&quot;&quot;<br/>    <br/>        client = irc_session.get_client(nodelay = True)<br/>    <br/>        client.send_raw([&quot;NICK nick4&quot;,<br/>                         &quot;2\r\n&quot;,<br/>                         &quot;USER user4&quot;,<br/>                         &quot;2 * * :User Forty Two\r\n&quot;],<br/>                        wait=0.05)<br/>    <br/>        irc_session.get_reply(client, expect_code = replies.RPL_WELCOME, expect_nick=&quot;nick42&quot;, expect_nparams = 1,<br/>&gt;                             long_param_re= &quot;Welcome to the Internet Relay Network nick42!user42@.*&quot;)<br/><br/>tests/chirc/tests/test_connection.py:218: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/>tests/chirc/tests/common.py:356: in get_reply<br/>    self.verify_reply(msg, expect_code, expect_nick, expect_nparams, expect_short_params, long_param_re, long_param_values)<br/>tests/chirc/tests/common.py:431: in verify_reply<br/>    long_param_re = long_param_re, long_param_values = long_param_values)<br/>tests/chirc/tests/common.py:404: in verify_message<br/>    self._assert_is_not_none(match, &quot;|||%s||| &lt;-- Long parameter does not match regular expression: %s&quot; % (lp, lpre), irc_msg = msg)<br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104e17a20&gt;, a = None<br/>explanation = &#x27;|||:Welcome to the Internet Relay Network nick42!nick42@localhost||| &lt;-- Long parameter does not match regular expres...2@.*$\n\nMESSAGE: |||:MacBook-Pro-5.local 001 nick42 :Welcome to the Internet Relay Network nick42!nick42@localhost|||&#x27;<br/>irc_msg = &lt;chirc.types.IRCMessage object at 0x104e43d30&gt;<br/><br/>    def _assert_is_not_none(self, a, explanation, irc_msg = None):<br/>        if irc_msg is not None:<br/>            explanation = explanation + &quot;\n\nMESSAGE: {}&quot;.format(irc_msg.raw(bookends=True))<br/>    <br/>&gt;       assert a is not None, explanation<br/><span class="error">E       AssertionError: |||:Welcome to the Internet Relay Network nick42!nick42@localhost||| &lt;-- Long parameter does not match regular expression: ^:Welcome to the Internet Relay Network nick42!user42@.*$</span><br/><span class="error">E       </span><br/><span class="error">E       MESSAGE: |||:MacBook-Pro-5.local 001 nick42 :Welcome to the Internet Relay Network nick42!nick42@localhost|||</span><br/><br/>tests/chirc/tests/common.py:56: AssertionError<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_nick_user_parsing3</td>
          <td class="col-duration">0.53</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.test_connection.TestBasicConnection object at 0x104e1d2e8&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104e1dba8&gt;<br/><br/>    def test_connect_nick_user_parsing3(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Tests that the server is actually parsing the NICK and USER parameters<br/>            (as opposed to just hardcoding &quot;nick1&quot; and &quot;user1&quot; which is used in<br/>            other tests. Additionally, partitions the messages.<br/>            &quot;&quot;&quot;<br/>    <br/>        client = irc_session.get_client(nodelay = True)<br/>    <br/>        client.send_raw([&quot;NICK nick4&quot;,<br/>                         &quot;2&quot;,<br/>                         &quot;4&quot;,<br/>                         &quot;2&quot;,<br/>                         &quot;\r\n&quot;,<br/>                         &quot;USER user4&quot;,<br/>                         &quot;2&quot;,<br/>                         &quot;4&quot;,<br/>                         &quot;2&quot;,<br/>                         &quot; * * :User Four Thousand Two Hundred and Forty Two\r\n&quot;],<br/>                        wait=0.05)<br/>    <br/>        irc_session.get_reply(client, expect_code = replies.RPL_WELCOME, expect_nick=&quot;nick4242&quot;, expect_nparams = 1,<br/>&gt;                             long_param_re= &quot;Welcome to the Internet Relay Network nick4242!user4242@.*&quot;)<br/><br/>tests/chirc/tests/test_connection.py:242: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/>tests/chirc/tests/common.py:356: in get_reply<br/>    self.verify_reply(msg, expect_code, expect_nick, expect_nparams, expect_short_params, long_param_re, long_param_values)<br/>tests/chirc/tests/common.py:431: in verify_reply<br/>    long_param_re = long_param_re, long_param_values = long_param_values)<br/>tests/chirc/tests/common.py:404: in verify_message<br/>    self._assert_is_not_none(match, &quot;|||%s||| &lt;-- Long parameter does not match regular expression: %s&quot; % (lp, lpre), irc_msg = msg)<br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104e1dba8&gt;, a = None<br/>explanation = &#x27;|||:Welcome to the Internet Relay Network nick4242!nick4242@localhost||| &lt;-- Long parameter does not match regular ex...n\nMESSAGE: |||:MacBook-Pro-5.local 001 nick4242 :Welcome to the Internet Relay Network nick4242!nick4242@localhost|||&#x27;<br/>irc_msg = &lt;chirc.types.IRCMessage object at 0x104e1d908&gt;<br/><br/>    def _assert_is_not_none(self, a, explanation, irc_msg = None):<br/>        if irc_msg is not None:<br/>            explanation = explanation + &quot;\n\nMESSAGE: {}&quot;.format(irc_msg.raw(bookends=True))<br/>    <br/>&gt;       assert a is not None, explanation<br/><span class="error">E       AssertionError: |||:Welcome to the Internet Relay Network nick4242!nick4242@localhost||| &lt;-- Long parameter does not match regular expression: ^:Welcome to the Internet Relay Network nick4242!user4242@.*$</span><br/><span class="error">E       </span><br/><span class="error">E       MESSAGE: |||:MacBook-Pro-5.local 001 nick4242 :Welcome to the Internet Relay Network nick4242!nick4242@localhost|||</span><br/><br/>tests/chirc/tests/common.py:56: AssertionError<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_nick_user_parsing4</td>
          <td class="col-duration">0.27</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.test_connection.TestBasicConnection object at 0x104e196d8&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104e3dd68&gt;<br/><br/>    def test_connect_nick_user_parsing4(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Tests that the server only uses the last NICK message sent during<br/>            the registration phase.<br/>            &quot;&quot;&quot;<br/>    <br/>        client = irc_session.get_client(nodelay = True)<br/>    <br/>        client.send_raw([&quot;NICK nick1\r\n&quot;,<br/>                         &quot;NICK nick2\r\n&quot;,<br/>                         &quot;NICK nick3\r\n&quot;,<br/>                         &quot;NICK nick42\r\n&quot;,<br/>                         &quot;USER user42 * * :User Forty Two\r\n&quot;],<br/>                        wait=0.05)<br/>    <br/>        irc_session.get_reply(client, expect_code = replies.RPL_WELCOME, expect_nick=&quot;nick42&quot;, expect_nparams = 1,<br/>&gt;                             long_param_re= &quot;Welcome to the Internet Relay Network nick42!user42@.*&quot;)<br/><br/>tests/chirc/tests/test_connection.py:260: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/>tests/chirc/tests/common.py:356: in get_reply<br/>    self.verify_reply(msg, expect_code, expect_nick, expect_nparams, expect_short_params, long_param_re, long_param_values)<br/>tests/chirc/tests/common.py:431: in verify_reply<br/>    long_param_re = long_param_re, long_param_values = long_param_values)<br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104e3dd68&gt;<br/>msg = &lt;chirc.types.IRCMessage object at 0x104e3dfd0&gt;, expect_prefix = True<br/>expect_cmd = &#x27;001&#x27;, expect_nparams = 2, expect_short_params = [&#x27;nick42&#x27;]<br/>long_param_re = &#x27;Welcome to the Internet Relay Network nick42!user42@.*&#x27;<br/>long_param_values = None<br/><br/>    def verify_message(self, msg, expect_prefix = None, expect_cmd = None,<br/>                      expect_nparams = None, expect_short_params = None,<br/>                      long_param_re = None, long_param_values = None):<br/>    <br/>        if expect_prefix != None and expect_prefix:<br/>&gt;           assert msg.prefix is not None, &quot;Expected a prefix, but got none.\nMessage: {}&quot;.format(msg.raw(bookends=True))<br/><span class="error">E           AssertionError: Expected a prefix, but got none.</span><br/><span class="error">E           Message: |||error, nickname already in use|||</span><br/><br/>tests/chirc/tests/common.py:380: AssertionError<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_nick_user_parsing5</td>
          <td class="col-duration">0.28</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.test_connection.TestBasicConnection object at 0x104e2b7b8&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104e2b828&gt;<br/><br/>    def test_connect_nick_user_parsing5(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Tests that the server only uses the last USER message sent during<br/>            the registration phase.<br/>            &quot;&quot;&quot;<br/>    <br/>        client = irc_session.get_client(nodelay = True)<br/>    <br/>        client.send_raw([&quot;USER user1 * * :User One\r\n&quot;,<br/>                         &quot;USER user2 * * :User Two\r\n&quot;,<br/>                         &quot;USER user3 * * :User Three\r\n&quot;,<br/>                         &quot;USER user42 * * :User Forty Two\r\n&quot;,<br/>                         &quot;NICK nick42\r\n&quot;,<br/>                         ],<br/>                        wait=0.05)<br/>    <br/>        irc_session.get_reply(client, expect_code = replies.RPL_WELCOME, expect_nick=&quot;nick42&quot;, expect_nparams = 1,<br/>&gt;                             long_param_re= &quot;Welcome to the Internet Relay Network nick42!user42@.*&quot;)<br/><br/>tests/chirc/tests/test_connection.py:279: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/>tests/chirc/tests/common.py:356: in get_reply<br/>    self.verify_reply(msg, expect_code, expect_nick, expect_nparams, expect_short_params, long_param_re, long_param_values)<br/>tests/chirc/tests/common.py:431: in verify_reply<br/>    long_param_re = long_param_re, long_param_values = long_param_values)<br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104e2b828&gt;<br/>msg = &lt;chirc.types.IRCMessage object at 0x104e52978&gt;, expect_prefix = True<br/>expect_cmd = &#x27;001&#x27;, expect_nparams = 2, expect_short_params = [&#x27;nick42&#x27;]<br/>long_param_re = &#x27;Welcome to the Internet Relay Network nick42!user42@.*&#x27;<br/>long_param_values = None<br/><br/>    def verify_message(self, msg, expect_prefix = None, expect_cmd = None,<br/>                      expect_nparams = None, expect_short_params = None,<br/>                      long_param_re = None, long_param_values = None):<br/>    <br/>        if expect_prefix != None and expect_prefix:<br/>&gt;           assert msg.prefix is not None, &quot;Expected a prefix, but got none.\nMessage: {}&quot;.format(msg.raw(bookends=True))<br/><span class="error">E           AssertionError: Expected a prefix, but got none.</span><br/><span class="error">E           Message: |||error, nickname already in use|||</span><br/><br/>tests/chirc/tests/common.py:380: AssertionError<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_no_unexpected_welcome4</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.test_connection.TestBasicConnection object at 0x104e3d0b8&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104e962b0&gt;<br/><br/>    def test_connect_no_unexpected_welcome4(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Sends two USER commands, but no NICK, so no RPL_WELCOME should be sent.<br/>            (tests that the server isn&#x27;t just unconditionally sending the welcome<br/>            messages after receiving two messages)<br/>            &quot;&quot;&quot;<br/>    <br/>        client = irc_session.get_client(nodelay = True)<br/>    <br/>        client.send_raw([&quot;USER user1 * * :User One\r\n&quot;,<br/>                         &quot;USER user2 * * :User Two\r\n&quot;],<br/>                        wait=0.05)<br/>    <br/>&gt;       irc_session.get_reply(client, expect_timeout = True)<br/><br/>tests/chirc/tests/test_connection.py:333: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104e962b0&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104e96128&gt;, expect_code = None<br/>expect_nick = None, expect_nparams = None, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = True<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>            msg = client.get_message()<br/>    <br/>            if expect_timeout:<br/>&gt;               pytest.fail(&quot;Was not expecting a reply, but got one:\n&quot; + msg.raw(bookends=True))<br/><span class="error">E               Failed: Was not expecting a reply, but got one:</span><br/><span class="error">E               |||error, nickname already in use|||</span><br/><br/>tests/chirc/tests/common.py:343: Failed<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_no_unexpected_welcome5</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.test_connection.TestBasicConnection object at 0x104ea7fd0&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104e78da0&gt;<br/><br/>    def test_connect_no_unexpected_welcome5(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Sends two NICK commands, but no USER, so no RPL_WELCOME should be sent.<br/>            (tests that the server isn&#x27;t just unconditionally sending the welcome<br/>            messages after receiving two messages)<br/>            &quot;&quot;&quot;<br/>    <br/>        client = irc_session.get_client(nodelay = True)<br/>    <br/>        client.send_raw([&quot;NICK user1\r\n&quot;,<br/>                         &quot;NICK user2\r\n&quot;],<br/>                        wait=0.05)<br/>    <br/>&gt;       irc_session.get_reply(client, expect_timeout = True)<br/><br/>tests/chirc/tests/test_connection.py:348: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104e78da0&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104e78128&gt;, expect_code = None<br/>expect_nick = None, expect_nparams = None, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = True<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>            msg = client.get_message()<br/>    <br/>            if expect_timeout:<br/>&gt;               pytest.fail(&quot;Was not expecting a reply, but got one:\n&quot; + msg.raw(bookends=True))<br/><span class="error">E               Failed: Was not expecting a reply, but got one:</span><br/><span class="error">E               |||error, nickname already in use|||</span><br/><br/>tests/chirc/tests/common.py:343: Failed<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_no_unexpected_welcome6</td>
          <td class="col-duration">0.27</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.test_connection.TestBasicConnection object at 0x104ea76d8&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x101cdbfd0&gt;<br/><br/>    def test_connect_no_unexpected_welcome6(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Sends multiple USER commands, but no NICK. No RPL_WELCOME should be sent.<br/>            &quot;&quot;&quot;<br/>    <br/>        client = irc_session.get_client(nodelay = True)<br/>    <br/>        client.send_raw([&quot;USER user1 * * :User One\r\n&quot;,<br/>                         &quot;USER user2 * * :User Two\r\n&quot;,<br/>                         &quot;USER user3 * * :User Two\r\n&quot;,<br/>                         &quot;USER user4 * * :User Two\r\n&quot;,<br/>                         &quot;USER user5 * * :User Two\r\n&quot;,],<br/>                        wait=0.05)<br/>    <br/>&gt;       irc_session.get_reply(client, expect_timeout = True)<br/><br/>tests/chirc/tests/test_connection.py:364: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x101cdbfd0&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104e19240&gt;, expect_code = None<br/>expect_nick = None, expect_nparams = None, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = True<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>            msg = client.get_message()<br/>    <br/>            if expect_timeout:<br/>&gt;               pytest.fail(&quot;Was not expecting a reply, but got one:\n&quot; + msg.raw(bookends=True))<br/><span class="error">E               Failed: Was not expecting a reply, but got one:</span><br/><span class="error">E               |||error, nickname already in use|||</span><br/><br/>tests/chirc/tests/common.py:343: Failed<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_no_unexpected_welcome7</td>
          <td class="col-duration">0.27</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.test_connection.TestBasicConnection object at 0x104e6e2e8&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104e6e048&gt;<br/><br/>    def test_connect_no_unexpected_welcome7(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Sends multiple NICK commands, but no USER. No RPL_WELCOME should be sent.<br/>            &quot;&quot;&quot;<br/>    <br/>        client = irc_session.get_client(nodelay = True)<br/>    <br/>        client.send_raw([&quot;NICK user1\r\n&quot;,<br/>                         &quot;NICK user2\r\n&quot;,<br/>                         &quot;NICK user3\r\n&quot;,<br/>                         &quot;NICK user4\r\n&quot;,<br/>                         &quot;NICK user5\r\n&quot;],<br/>                        wait=0.05)<br/>    <br/>&gt;       irc_session.get_reply(client, expect_timeout = True)<br/><br/>tests/chirc/tests/test_connection.py:380: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104e6e048&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104e6e3c8&gt;, expect_code = None<br/>expect_nick = None, expect_nparams = None, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = True<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>            msg = client.get_message()<br/>    <br/>            if expect_timeout:<br/>&gt;               pytest.fail(&quot;Was not expecting a reply, but got one:\n&quot; + msg.raw(bookends=True))<br/><span class="error">E               Failed: Was not expecting a reply, but got one:</span><br/><span class="error">E               |||error, nickname already in use|||</span><br/><br/>tests/chirc/tests/common.py:343: Failed<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestFullConnection::()::test_connect_full1</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.common.IRCSession object at 0x104e52fd0&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104e52048&gt;, expect_code = &#x27;001&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = &#x27;Welcome to the Internet Relay Network user1!user1.*&#x27;<br/>long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>&gt;           msg = client.get_message()<br/><br/>tests/chirc/tests/common.py:340: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.client.ChircClient object at 0x104e52048&gt;<br/><br/>    def get_message(self):<br/>        msg = self.client.read_until(str.encode(&quot;\r\n&quot;), timeout=self.msg_timeout)<br/>        msg = msg.decode()<br/>        if msg[-2:] != &quot;\r\n&quot;:<br/>&gt;           raise ReplyTimeoutException(msg)<br/><span class="error">E           chirc.types.ReplyTimeoutException</span><br/><br/>tests/chirc/client.py:37: ReplyTimeoutException<br/><br/>During handling of the above exception, another exception occurred:<br/><br/>self = &lt;chirc.tests.test_connection.TestFullConnection object at 0x104e522b0&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104e52fd0&gt;<br/><br/>    def test_connect_full1(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Checks that all welcome messages are sent after the NICK and USER<br/>            commands are received.<br/>            &quot;&quot;&quot;<br/>    <br/>        client = irc_session.get_client()<br/>    <br/>        client.send_cmd(&quot;NICK user1&quot;)<br/>        client.send_cmd(&quot;USER user1 * * :User One&quot;)<br/>    <br/>&gt;       irc_session.verify_welcome_messages(client, &quot;user1&quot;)<br/><br/>tests/chirc/tests/test_connection.py:397: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/>tests/chirc/tests/common.py:441: in verify_welcome_messages<br/>    long_param_re= &quot;Welcome to the Internet Relay Network {}!{}.*&quot;.format(nick, user))<br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104e52fd0&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104e52048&gt;, expect_code = &#x27;001&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = &#x27;Welcome to the Internet Relay Network user1!user1.*&#x27;<br/>long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>            msg = client.get_message()<br/>    <br/>            if expect_timeout:<br/>                pytest.fail(&quot;Was not expecting a reply, but got one:\n&quot; + msg.raw(bookends=True))<br/>        except EOFError:<br/>            pytest.fail(&quot;Server closed connection unexpectedly. Possible segfault in server?&quot;)<br/>        except ReplyTimeoutException as rte:<br/>            if expect_timeout:<br/>                return None<br/>    <br/>            if len(rte.bytes_received) == 0:<br/>                failmsg = &quot;Expected a reply but got none (no bytes received)&quot;<br/>            else:<br/>                failmsg = &quot;Expected a reply but did not get valid reply terminated with \\r\\n. Bytes received:\n|||{}|||&quot;.format(rte.bytes_received)<br/>&gt;           pytest.fail(failmsg)<br/><span class="error">E           Failed: Expected a reply but got none (no bytes received)</span><br/><br/>tests/chirc/tests/common.py:354: Failed<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestFullConnection::()::test_connect_full2</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.common.IRCSession object at 0x104e7f160&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104e7f8d0&gt;, expect_code = &#x27;001&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = &#x27;Welcome to the Internet Relay Network user1!user1.*&#x27;<br/>long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>&gt;           msg = client.get_message()<br/><br/>tests/chirc/tests/common.py:340: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.client.ChircClient object at 0x104e7f8d0&gt;<br/><br/>    def get_message(self):<br/>        msg = self.client.read_until(str.encode(&quot;\r\n&quot;), timeout=self.msg_timeout)<br/>        msg = msg.decode()<br/>        if msg[-2:] != &quot;\r\n&quot;:<br/>&gt;           raise ReplyTimeoutException(msg)<br/><span class="error">E           chirc.types.ReplyTimeoutException</span><br/><br/>tests/chirc/client.py:37: ReplyTimeoutException<br/><br/>During handling of the above exception, another exception occurred:<br/><br/>self = &lt;chirc.tests.test_connection.TestFullConnection object at 0x104e7f710&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104e7f160&gt;<br/><br/>    def test_connect_full2(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Checks that all welcome messages are sent after the USER and NICK<br/>            commands are received.<br/>            &quot;&quot;&quot;<br/>    <br/>        client = irc_session.get_client()<br/>    <br/>        client.send_cmd(&quot;USER user1 * * :User One&quot;)<br/>        client.send_cmd(&quot;NICK user1&quot;)<br/>    <br/>&gt;       irc_session.verify_welcome_messages(client, &quot;user1&quot;)<br/><br/>tests/chirc/tests/test_connection.py:410: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/>tests/chirc/tests/common.py:441: in verify_welcome_messages<br/>    long_param_re= &quot;Welcome to the Internet Relay Network {}!{}.*&quot;.format(nick, user))<br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104e7f160&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104e7f8d0&gt;, expect_code = &#x27;001&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = &#x27;Welcome to the Internet Relay Network user1!user1.*&#x27;<br/>long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>            msg = client.get_message()<br/>    <br/>            if expect_timeout:<br/>                pytest.fail(&quot;Was not expecting a reply, but got one:\n&quot; + msg.raw(bookends=True))<br/>        except EOFError:<br/>            pytest.fail(&quot;Server closed connection unexpectedly. Possible segfault in server?&quot;)<br/>        except ReplyTimeoutException as rte:<br/>            if expect_timeout:<br/>                return None<br/>    <br/>            if len(rte.bytes_received) == 0:<br/>                failmsg = &quot;Expected a reply but got none (no bytes received)&quot;<br/>            else:<br/>                failmsg = &quot;Expected a reply but did not get valid reply terminated with \\r\\n. Bytes received:\n|||{}|||&quot;.format(rte.bytes_received)<br/>&gt;           pytest.fail(failmsg)<br/><span class="error">E           Failed: Expected a reply but got none (no bytes received)</span><br/><br/>tests/chirc/tests/common.py:354: Failed<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestFullConnection::()::test_connect_full3</td>
          <td class="col-duration">0.10</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.common.IRCSession object at 0x104d01160&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104d011d0&gt;, expect_code = &#x27;001&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = &#x27;Welcome to the Internet Relay Network user1!user1.*&#x27;<br/>long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>&gt;           msg = client.get_message()<br/><br/>tests/chirc/tests/common.py:340: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.client.ChircClient object at 0x104d011d0&gt;<br/><br/>    def get_message(self):<br/>        msg = self.client.read_until(str.encode(&quot;\r\n&quot;), timeout=self.msg_timeout)<br/>        msg = msg.decode()<br/>        if msg[-2:] != &quot;\r\n&quot;:<br/>&gt;           raise ReplyTimeoutException(msg)<br/><span class="error">E           chirc.types.ReplyTimeoutException</span><br/><br/>tests/chirc/client.py:37: ReplyTimeoutException<br/><br/>During handling of the above exception, another exception occurred:<br/><br/>self = &lt;chirc.tests.test_connection.TestFullConnection object at 0x104aad358&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104d01160&gt;<br/><br/>    def test_connect_full3(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Checks that all welcome messages, as well as the LUSERS and<br/>            MOTD replies, are sent after the NICK and USER commands are received.<br/>            &quot;&quot;&quot;<br/>    <br/>        client = irc_session.get_client()<br/>    <br/>        client.send_cmd(&quot;NICK user1&quot;)<br/>        client.send_cmd(&quot;USER user1 * * :User One&quot;)<br/>    <br/>&gt;       irc_session.verify_welcome_messages(client, &quot;user1&quot;)<br/><br/>tests/chirc/tests/test_connection.py:423: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/>tests/chirc/tests/common.py:441: in verify_welcome_messages<br/>    long_param_re= &quot;Welcome to the Internet Relay Network {}!{}.*&quot;.format(nick, user))<br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104d01160&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104d011d0&gt;, expect_code = &#x27;001&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = &#x27;Welcome to the Internet Relay Network user1!user1.*&#x27;<br/>long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>            msg = client.get_message()<br/>    <br/>            if expect_timeout:<br/>                pytest.fail(&quot;Was not expecting a reply, but got one:\n&quot; + msg.raw(bookends=True))<br/>        except EOFError:<br/>            pytest.fail(&quot;Server closed connection unexpectedly. Possible segfault in server?&quot;)<br/>        except ReplyTimeoutException as rte:<br/>            if expect_timeout:<br/>                return None<br/>    <br/>            if len(rte.bytes_received) == 0:<br/>                failmsg = &quot;Expected a reply but got none (no bytes received)&quot;<br/>            else:<br/>                failmsg = &quot;Expected a reply but did not get valid reply terminated with \\r\\n. Bytes received:\n|||{}|||&quot;.format(rte.bytes_received)<br/>&gt;           pytest.fail(failmsg)<br/><span class="error">E           Failed: Expected a reply but got none (no bytes received)</span><br/><br/>tests/chirc/tests/common.py:354: Failed<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestMultiuserConnection::()::test_connect_2users</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.common.IRCSession object at 0x104d0b710&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104d0b898&gt;, expect_code = &#x27;001&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = &#x27;Welcome to the Internet Relay Network user1!user1.*&#x27;<br/>long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>&gt;           msg = client.get_message()<br/><br/>tests/chirc/tests/common.py:340: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.client.ChircClient object at 0x104d0b898&gt;<br/><br/>    def get_message(self):<br/>        msg = self.client.read_until(str.encode(&quot;\r\n&quot;), timeout=self.msg_timeout)<br/>        msg = msg.decode()<br/>        if msg[-2:] != &quot;\r\n&quot;:<br/>&gt;           raise ReplyTimeoutException(msg)<br/><span class="error">E           chirc.types.ReplyTimeoutException</span><br/><br/>tests/chirc/client.py:37: ReplyTimeoutException<br/><br/>During handling of the above exception, another exception occurred:<br/><br/>self = &lt;chirc.tests.test_connection.TestMultiuserConnection object at 0x104d0b5c0&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104d0b710&gt;<br/><br/>    def test_connect_2users(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Connects two clients to the server.<br/>            &quot;&quot;&quot;<br/>    <br/>&gt;       irc_session.connect_user(&quot;user1&quot;, &quot;User One&quot;)<br/><br/>tests/chirc/tests/test_connection.py:436: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/>tests/chirc/tests/common.py:136: in connect_user<br/>    self.verify_welcome_messages(client, nick)<br/>tests/chirc/tests/common.py:441: in verify_welcome_messages<br/>    long_param_re= &quot;Welcome to the Internet Relay Network {}!{}.*&quot;.format(nick, user))<br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104d0b710&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104d0b898&gt;, expect_code = &#x27;001&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = &#x27;Welcome to the Internet Relay Network user1!user1.*&#x27;<br/>long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>            msg = client.get_message()<br/>    <br/>            if expect_timeout:<br/>                pytest.fail(&quot;Was not expecting a reply, but got one:\n&quot; + msg.raw(bookends=True))<br/>        except EOFError:<br/>            pytest.fail(&quot;Server closed connection unexpectedly. Possible segfault in server?&quot;)<br/>        except ReplyTimeoutException as rte:<br/>            if expect_timeout:<br/>                return None<br/>    <br/>            if len(rte.bytes_received) == 0:<br/>                failmsg = &quot;Expected a reply but got none (no bytes received)&quot;<br/>            else:<br/>                failmsg = &quot;Expected a reply but did not get valid reply terminated with \\r\\n. Bytes received:\n|||{}|||&quot;.format(rte.bytes_received)<br/>&gt;           pytest.fail(failmsg)<br/><span class="error">E           Failed: Expected a reply but got none (no bytes received)</span><br/><br/>tests/chirc/tests/common.py:354: Failed<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestMultiuserConnection::()::test_connect_duplicate_nick</td>
          <td class="col-duration">0.10</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.common.IRCSession object at 0x104b27588&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104b27080&gt;, expect_code = &#x27;001&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = &#x27;Welcome to the Internet Relay Network user1!user1.*&#x27;<br/>long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>&gt;           msg = client.get_message()<br/><br/>tests/chirc/tests/common.py:340: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.client.ChircClient object at 0x104b27080&gt;<br/><br/>    def get_message(self):<br/>        msg = self.client.read_until(str.encode(&quot;\r\n&quot;), timeout=self.msg_timeout)<br/>        msg = msg.decode()<br/>        if msg[-2:] != &quot;\r\n&quot;:<br/>&gt;           raise ReplyTimeoutException(msg)<br/><span class="error">E           chirc.types.ReplyTimeoutException</span><br/><br/>tests/chirc/client.py:37: ReplyTimeoutException<br/><br/>During handling of the above exception, another exception occurred:<br/><br/>self = &lt;chirc.tests.test_connection.TestMultiuserConnection object at 0x104b274a8&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104b27588&gt;<br/><br/>    def test_connect_duplicate_nick(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Connects two clients to the server, but the second<br/>            client tries to use the first client&#x27;s nickname<br/>            (and should get an ERR_NICKNAMEINUSE)<br/>            &quot;&quot;&quot;<br/>    <br/>&gt;       client1 = irc_session.connect_user(&quot;user1&quot;, &quot;User One&quot;)<br/><br/>tests/chirc/tests/test_connection.py:446: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/>tests/chirc/tests/common.py:136: in connect_user<br/>    self.verify_welcome_messages(client, nick)<br/>tests/chirc/tests/common.py:441: in verify_welcome_messages<br/>    long_param_re= &quot;Welcome to the Internet Relay Network {}!{}.*&quot;.format(nick, user))<br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104b27588&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104b27080&gt;, expect_code = &#x27;001&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = &#x27;Welcome to the Internet Relay Network user1!user1.*&#x27;<br/>long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>            msg = client.get_message()<br/>    <br/>            if expect_timeout:<br/>                pytest.fail(&quot;Was not expecting a reply, but got one:\n&quot; + msg.raw(bookends=True))<br/>        except EOFError:<br/>            pytest.fail(&quot;Server closed connection unexpectedly. Possible segfault in server?&quot;)<br/>        except ReplyTimeoutException as rte:<br/>            if expect_timeout:<br/>                return None<br/>    <br/>            if len(rte.bytes_received) == 0:<br/>                failmsg = &quot;Expected a reply but got none (no bytes received)&quot;<br/>            else:<br/>                failmsg = &quot;Expected a reply but did not get valid reply terminated with \\r\\n. Bytes received:\n|||{}|||&quot;.format(rte.bytes_received)<br/>&gt;           pytest.fail(failmsg)<br/><span class="error">E           Failed: Expected a reply but got none (no bytes received)</span><br/><br/>tests/chirc/tests/common.py:354: Failed<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestQUIT::()::test_quit_after_registration1</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.common.IRCSession object at 0x104b502b0&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104b50550&gt;, expect_code = &#x27;001&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = &#x27;Welcome to the Internet Relay Network user1!user1.*&#x27;<br/>long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>&gt;           msg = client.get_message()<br/><br/>tests/chirc/tests/common.py:340: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.client.ChircClient object at 0x104b50550&gt;<br/><br/>    def get_message(self):<br/>        msg = self.client.read_until(str.encode(&quot;\r\n&quot;), timeout=self.msg_timeout)<br/>        msg = msg.decode()<br/>        if msg[-2:] != &quot;\r\n&quot;:<br/>&gt;           raise ReplyTimeoutException(msg)<br/><span class="error">E           chirc.types.ReplyTimeoutException</span><br/><br/>tests/chirc/client.py:37: ReplyTimeoutException<br/><br/>During handling of the above exception, another exception occurred:<br/><br/>self = &lt;chirc.tests.test_connection.TestQUIT object at 0x104b503c8&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104b502b0&gt;<br/><br/>    def test_quit_after_registration1(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Connects a client, and quits right after connecting.<br/>            Verifies the ERROR response, but not whether the server<br/>            actually disconnected the client.<br/>            &quot;&quot;&quot;<br/>    <br/>&gt;       client1 = irc_session.connect_user(&quot;user1&quot;, &quot;User One&quot;)<br/><br/>tests/chirc/tests/test_connection.py:465: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/>tests/chirc/tests/common.py:136: in connect_user<br/>    self.verify_welcome_messages(client, nick)<br/>tests/chirc/tests/common.py:441: in verify_welcome_messages<br/>    long_param_re= &quot;Welcome to the Internet Relay Network {}!{}.*&quot;.format(nick, user))<br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104b502b0&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104b50550&gt;, expect_code = &#x27;001&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = &#x27;Welcome to the Internet Relay Network user1!user1.*&#x27;<br/>long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>            msg = client.get_message()<br/>    <br/>            if expect_timeout:<br/>                pytest.fail(&quot;Was not expecting a reply, but got one:\n&quot; + msg.raw(bookends=True))<br/>        except EOFError:<br/>            pytest.fail(&quot;Server closed connection unexpectedly. Possible segfault in server?&quot;)<br/>        except ReplyTimeoutException as rte:<br/>            if expect_timeout:<br/>                return None<br/>    <br/>            if len(rte.bytes_received) == 0:<br/>                failmsg = &quot;Expected a reply but got none (no bytes received)&quot;<br/>            else:<br/>                failmsg = &quot;Expected a reply but did not get valid reply terminated with \\r\\n. Bytes received:\n|||{}|||&quot;.format(rte.bytes_received)<br/>&gt;           pytest.fail(failmsg)<br/><span class="error">E           Failed: Expected a reply but got none (no bytes received)</span><br/><br/>tests/chirc/tests/common.py:354: Failed<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestQUIT::()::test_quit_after_registration2</td>
          <td class="col-duration">0.10</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.common.IRCSession object at 0x104e78dd8&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104e52b38&gt;, expect_code = &#x27;001&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = &#x27;Welcome to the Internet Relay Network user1!user1.*&#x27;<br/>long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>&gt;           msg = client.get_message()<br/><br/>tests/chirc/tests/common.py:340: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.client.ChircClient object at 0x104e52b38&gt;<br/><br/>    def get_message(self):<br/>        msg = self.client.read_until(str.encode(&quot;\r\n&quot;), timeout=self.msg_timeout)<br/>        msg = msg.decode()<br/>        if msg[-2:] != &quot;\r\n&quot;:<br/>&gt;           raise ReplyTimeoutException(msg)<br/><span class="error">E           chirc.types.ReplyTimeoutException</span><br/><br/>tests/chirc/client.py:37: ReplyTimeoutException<br/><br/>During handling of the above exception, another exception occurred:<br/><br/>self = &lt;chirc.tests.test_connection.TestQUIT object at 0x104e78b00&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104e78dd8&gt;<br/><br/>    def test_quit_after_registration2(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Connects a client, and quits right after connecting.<br/>            Verifies the ERROR response and also checks whether the<br/>            server disconnected the client.<br/>            &quot;&quot;&quot;<br/>    <br/>&gt;       client1 = irc_session.connect_user(&quot;user1&quot;, &quot;User One&quot;)<br/><br/>tests/chirc/tests/test_connection.py:479: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/>tests/chirc/tests/common.py:136: in connect_user<br/>    self.verify_welcome_messages(client, nick)<br/>tests/chirc/tests/common.py:441: in verify_welcome_messages<br/>    long_param_re= &quot;Welcome to the Internet Relay Network {}!{}.*&quot;.format(nick, user))<br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104e78dd8&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104e52b38&gt;, expect_code = &#x27;001&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = &#x27;Welcome to the Internet Relay Network user1!user1.*&#x27;<br/>long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>            msg = client.get_message()<br/>    <br/>            if expect_timeout:<br/>                pytest.fail(&quot;Was not expecting a reply, but got one:\n&quot; + msg.raw(bookends=True))<br/>        except EOFError:<br/>            pytest.fail(&quot;Server closed connection unexpectedly. Possible segfault in server?&quot;)<br/>        except ReplyTimeoutException as rte:<br/>            if expect_timeout:<br/>                return None<br/>    <br/>            if len(rte.bytes_received) == 0:<br/>                failmsg = &quot;Expected a reply but got none (no bytes received)&quot;<br/>            else:<br/>                failmsg = &quot;Expected a reply but did not get valid reply terminated with \\r\\n. Bytes received:\n|||{}|||&quot;.format(rte.bytes_received)<br/>&gt;           pytest.fail(failmsg)<br/><span class="error">E           Failed: Expected a reply but got none (no bytes received)</span><br/><br/>tests/chirc/tests/common.py:354: Failed<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestQUIT::()::test_quit_after_registration3</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.common.IRCSession object at 0x104d0b208&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104d0b3c8&gt;, expect_code = &#x27;001&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = &#x27;Welcome to the Internet Relay Network user1!user1.*&#x27;<br/>long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>&gt;           msg = client.get_message()<br/><br/>tests/chirc/tests/common.py:340: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.client.ChircClient object at 0x104d0b3c8&gt;<br/><br/>    def get_message(self):<br/>        msg = self.client.read_until(str.encode(&quot;\r\n&quot;), timeout=self.msg_timeout)<br/>        msg = msg.decode()<br/>        if msg[-2:] != &quot;\r\n&quot;:<br/>&gt;           raise ReplyTimeoutException(msg)<br/><span class="error">E           chirc.types.ReplyTimeoutException</span><br/><br/>tests/chirc/client.py:37: ReplyTimeoutException<br/><br/>During handling of the above exception, another exception occurred:<br/><br/>self = &lt;chirc.tests.test_connection.TestQUIT object at 0x104d0bac8&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104d0b208&gt;<br/><br/>    def test_quit_after_registration3(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Connects a client, and quits right after connecting.<br/>            QUIT does not specify a message, so the default &quot;Client Quit&quot; should<br/>            be assumed. Also verifies the ERROR response and also checks whether the<br/>            server disconnected the client.<br/>            &quot;&quot;&quot;<br/>    <br/>&gt;       client1 = irc_session.connect_user(&quot;user1&quot;, &quot;User One&quot;)<br/><br/>tests/chirc/tests/test_connection.py:496: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/>tests/chirc/tests/common.py:136: in connect_user<br/>    self.verify_welcome_messages(client, nick)<br/>tests/chirc/tests/common.py:441: in verify_welcome_messages<br/>    long_param_re= &quot;Welcome to the Internet Relay Network {}!{}.*&quot;.format(nick, user))<br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104d0b208&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104d0b3c8&gt;, expect_code = &#x27;001&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = &#x27;Welcome to the Internet Relay Network user1!user1.*&#x27;<br/>long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>            msg = client.get_message()<br/>    <br/>            if expect_timeout:<br/>                pytest.fail(&quot;Was not expecting a reply, but got one:\n&quot; + msg.raw(bookends=True))<br/>        except EOFError:<br/>            pytest.fail(&quot;Server closed connection unexpectedly. Possible segfault in server?&quot;)<br/>        except ReplyTimeoutException as rte:<br/>            if expect_timeout:<br/>                return None<br/>    <br/>            if len(rte.bytes_received) == 0:<br/>                failmsg = &quot;Expected a reply but got none (no bytes received)&quot;<br/>            else:<br/>                failmsg = &quot;Expected a reply but did not get valid reply terminated with \\r\\n. Bytes received:\n|||{}|||&quot;.format(rte.bytes_received)<br/>&gt;           pytest.fail(failmsg)<br/><span class="error">E           Failed: Expected a reply but got none (no bytes received)</span><br/><br/>tests/chirc/tests/common.py:354: Failed<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestQUIT::()::test_quit_after_registration4</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.common.IRCSession object at 0x1060080f0&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x10600e240&gt;, expect_code = &#x27;001&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = &#x27;Welcome to the Internet Relay Network user1!user1.*&#x27;<br/>long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>&gt;           msg = client.get_message()<br/><br/>tests/chirc/tests/common.py:340: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.client.ChircClient object at 0x10600e240&gt;<br/><br/>    def get_message(self):<br/>        msg = self.client.read_until(str.encode(&quot;\r\n&quot;), timeout=self.msg_timeout)<br/>        msg = msg.decode()<br/>        if msg[-2:] != &quot;\r\n&quot;:<br/>&gt;           raise ReplyTimeoutException(msg)<br/><span class="error">E           chirc.types.ReplyTimeoutException</span><br/><br/>tests/chirc/client.py:37: ReplyTimeoutException<br/><br/>During handling of the above exception, another exception occurred:<br/><br/>self = &lt;chirc.tests.test_connection.TestQUIT object at 0x106008b38&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x1060080f0&gt;<br/><br/>    def test_quit_after_registration4(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Connects two clients, and then has them quit, each with separate<br/>            messages.<br/>            &quot;&quot;&quot;<br/>    <br/>&gt;       client1 = irc_session.connect_user(&quot;user1&quot;, &quot;User One&quot;)<br/><br/>tests/chirc/tests/test_connection.py:512: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/>tests/chirc/tests/common.py:136: in connect_user<br/>    self.verify_welcome_messages(client, nick)<br/>tests/chirc/tests/common.py:441: in verify_welcome_messages<br/>    long_param_re= &quot;Welcome to the Internet Relay Network {}!{}.*&quot;.format(nick, user))<br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x1060080f0&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x10600e240&gt;, expect_code = &#x27;001&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = &#x27;Welcome to the Internet Relay Network user1!user1.*&#x27;<br/>long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>            msg = client.get_message()<br/>    <br/>            if expect_timeout:<br/>                pytest.fail(&quot;Was not expecting a reply, but got one:\n&quot; + msg.raw(bookends=True))<br/>        except EOFError:<br/>            pytest.fail(&quot;Server closed connection unexpectedly. Possible segfault in server?&quot;)<br/>        except ReplyTimeoutException as rte:<br/>            if expect_timeout:<br/>                return None<br/>    <br/>            if len(rte.bytes_received) == 0:<br/>                failmsg = &quot;Expected a reply but got none (no bytes received)&quot;<br/>            else:<br/>                failmsg = &quot;Expected a reply but did not get valid reply terminated with \\r\\n. Bytes received:\n|||{}|||&quot;.format(rte.bytes_received)<br/>&gt;           pytest.fail(failmsg)<br/><span class="error">E           Failed: Expected a reply but got none (no bytes received)</span><br/><br/>tests/chirc/tests/common.py:354: Failed<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_simple1</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_partitioned_messages5</td>
          <td class="col-duration">0.21</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_partitioned_messages6</td>
          <td class="col-duration">0.27</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_no_unexpected_welcome1</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_no_unexpected_welcome2</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_no_unexpected_welcome3</td>
          <td class="col-duration">0.21</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody></table></body></html>